### ISSUE TITLE
Inconsistency when running the keras-io/examples/timeseries/timeseries_classification_transformer.py

### ISSUE THREAD

**Issue by condor-cp** [keras-team/keras-io#1908]
Running the example shows inconsistency in number of parameters and model performance compared to what is displayed.

It seems that the global average pooling should take data_format to "channel_first" to reach the same number of parameters and the accuracy performance consistent with the displayed console log (tried with google colab).

https://github.com/keras-team/keras-io/blob/2ac94c4c4da54c1f7c80fea7f18832fd54c6be28/examples/timeseries/timeseries_classification_transformer.py#L115

But then there is no pooling, just removing the feature dimension => Maybe another layer should be used.

**Comment by mw66** [comment]
Experienced the same problem:

with 
```
    x = layers.GlobalAveragePooling1D(data_format="channels_last")(x)                                                                                                                                              
```

the number of parameters are (only show the last few rows that differ):

```
...                                                                                                  
 global_average_pooling1d (Glob  (None, 1)           0           ['tf.__operators__.add_7[0][0]'] 
 alAveragePooling1D)                                                                              
                                                                                                  
 dense (Dense)                  (None, 128)          256         ['global_average_pooling1d[0][0]'
                                                                 ]                                
                                                                                                  
 dropout_8 (Dropout)            (None, 128)          0           ['dense[0][0]']                  
                                                                                                  
 dense_1 (Dense)                (None, 2)            258         ['dropout_8[0][0]']              
                                                                                                  
==================================================================================================
Total params: 29,258
Trainable params: 29,258
Non-trainable params: 0
```
And the training stop very quickly with bad result:
```
45/45 [==============================] - 24s 545ms/step - loss: 0.6922 - sparse_categorical_accuracy: 0.5208 - val_loss: 0.6952 - val_sparse_categorical_accuracy: 0.4799
42/42 [==============================] - 4s 91ms/step - loss: 0.6930 - sparse_categorical_accuracy: 0.5159
```


While on page: https://keras.io/examples/timeseries/timeseries_classification_transformer/

it shows:
```
│ global_average_poo… │ (None, 500)       │       0 │ add_7[0][0]          │
│ (GlobalAveragePool… │                   │         │                      │
├─────────────────────┼───────────────────┼─────────┼──────────────────────┤
│ dense (Dense)       │ (None, 128)       │  64,128 │ global_average_pool… │
├─────────────────────┼───────────────────┼─────────┼──────────────────────┤
│ dropout_12          │ (None, 128)       │       0 │ dense[0][0]          │
│ (Dropout)           │                   │         │                      │
├─────────────────────┼───────────────────┼─────────┼──────────────────────┤
│ dense_1 (Dense)     │ (None, 2)         │     258 │ dropout_12[0][0]     │
└─────────────────────┴───────────────────┴─────────┴──────────────────────┘
 Total params: 93,130 (363.79 KB)
 Trainable params: 93,130 (363.79 KB)
 Non-trainable params: 0 (0.00 B)
```

**Comment by mw66** [comment]
Hi , @fchollet 

```
https://github.com/keras-team/keras-io/blob/8ec940273419cd5a61f056f462753ba4ac0c9590/examples/timeseries/timeseries_classification_transformer.py#L115

```

This line need to be changed to `channel_first` to see the good result on:

https://keras.io/examples/timeseries/timeseries_classification_transformer/

Can you make this change? can also add explanation why `channel_first` is need instead of `channels_last`?

The input training data's shape is `(3601, 500, 1)`, i.e channels_last for sure; but why we need to set `channel_first` to see the good training result?

Thanks.

**Comment by mw66** [comment]
> This line need to be changed to `channel_first` to see the good result on:

or:
```
    # Reduces features to 1 while keeping 500 time steps
    x = layers.Conv1D(filters=1, kernel_size=1, activation='relu')(x)
    x = layers.Flatten()(x)
```

basically, need to keep the time steps.

**Comment by kharshith-k** [comment]
Hi, I dug into this to check whether it's a docs-regeneration issue or an actual bug in the code, and reproduced it directly.

**Shape/param verification** — I rebuilt `build_model()` from the current source using real Keras (Keras 3, torch backend) with both `data_format` settings:

- `data_format="channels_last"` (current code): GAP output `(None, 1)`, **29,258 total params** — matches the numbers `@mw66` posted above exactly.
- `data_format="channels_first"`: GAP output `(None, 500)`, **93,130 total params** — matches the numbers currently published on keras.io exactly.

This isn't a labeling/cosmetic difference. After the transformer blocks, the tensor entering `GlobalAveragePooling1D` has shape `(batch, 500, 1)` — 500 timesteps, 1 feature (the final `Conv1D` in `transformer_encoder` restores `filters=inputs.shape[-1]=1` for the residual add). `channels_last` correctly averages over the 500-step temporal axis, collapsing it to a single scalar per example. `channels_first` mislabels the axes and ends up averaging over the size-1 last axis instead — effectively a no-op that preserves the full 500-step vector.

**Training verification** — I also trained both variants end-to-end on the real FordA data (smaller config for speed: 2 transformer blocks, 600 train samples, 6 epochs) to check the reported accuracy claim rather than just take it on faith:

- `channels_last`: train/val accuracy stayed flat at ~0.51 for all 6 epochs — the model never learns anything, consistent with random guessing on this binary task.
- `channels_first`: train accuracy climbed 0.57 → 0.82 over the same 6 epochs, with corresponding validation accuracy gains.

So the root cause is real: collapsing all 500 timesteps to a single scalar (which is what `channels_last` correctly does here, given only 1 feature channel) destroys the signal needed for classification. `channels_first` "works" only because it accidentally skips real temporal pooling and passes the full 500-length representation to the `Dense` layers.

This means the fix isn't picking the `data_format` that reproduces the old published numbers — it's that `GlobalAveragePooling1D` is the wrong layer for this architecture given a single feature channel. Swapping it for `Flatten()` (or otherwise restructuring so pooling operates over a real, multi-dimensional feature axis) would be a more architecturally sound fix, though I haven't verified that specific change trains as well — happy to test that too if useful.
